internal/juju/applications.go

```diff
@@ -1413,6 +1413,9 @@ func (c applicationsClient) computeSetCharmConfig(
 	// is called.
 	if input.Revision != nil {
 		oldOrigin.Revision = input.Revision
+		// See comment at line 1368, the idea is to set these values to empty to trick Juju into getting the right revision.
```

Comment:
I wouldn't reference a line number because that will get outdated quickly.
Can you expand on the "trick" part. Maybe something like
`Set the values so that Juju uses the latest revision...`